### ISSUE TITLE
Fail catkin_make ros melodic [gazebo 9.0.0]

### ISSUE THREAD

**Issue by Yougo-robotics** [wojiaojiao/pegasus_gazebo_plugins#3]
The cpp file doesn't compile for gazebo with the ros melodic version due to some deprecated calls of functions such as `GetPhysicsEngine()` in the new Gazebo 9.0.0 versions.

- I fixed it so now it should work with gazebo 9.0.0
- I also added a functionality to don't have to go change `math::Pose jointOrigin(x,y,z,roll,pitch,yaw);` in the .cpp to change the orientation and the position because it was boring.

Now the you should be able to use this syntax instead of going inside the file 

>  <gazebo>
       <plugin name="Myrobot_ClosedLoopPlugin" filename="libpegasus_gazebo_closed_loop_plugin.so">
           <joint>Joint_name</joint>
           <child>Link_Child_name</child>
           <parent>Link_Parent_Name</parent>
           <rotation>0 0 0</rotation>
           <position>0.00 -0.00 0.00</position>
       </plugin>
     </gazebo>

Here is the plugin version for gazebo 9.0.0, it compiles but I can't currently fully test it so if someone can confirm that it's working it could be nice.

[pegasus_gazebo_plugins_Gv9.zip](https://github.com/wojiaojiao/pegasus_gazebo_plugins/files/3475518/pegasus_gazebo_plugins_Gv9.zip)

**Comment by mnbf9rca** [comment]
Thanks for updating this! i can't seem to get this to actually create the joint though. Have you got it working? when i inspect in gazebo, there's no joint created. here's the URDF https://github.com/mnbf9rca/simple_robot_arm_with_ros/blob/master/ros/src/simple_arm/urdf/simple_arm_loop.xacro

**Comment by wojiaojiao** [comment]
Thank you for your improvement @Yougo-robotics

**Comment by Ugo31** [comment]
Yes I got it running in gazebo 9, seems everything work physically talking. The joint have the expected behaviour, but I also couldn't see joints generated when I inspect it in gazebo

**Comment by mnbf9rca** [comment]
the joint didnt have the behaviour i expected in gazebo i.e. the parts physically separated when moving.

**Comment by Timple** [comment]
I have the same issue as @mnbf9rca The links are seperated (by approx 1m in my simulation). The do seem to be connected.

**Comment by wojiaojiao** [comment]
@Yougo-robotics 
There is an error in your code：
         ignition::math::Pose3d jointOrigin = ignition::math::Pose3d(
                                                                        rotations_splited_converted[0],
                                                                        rotations_splited_converted[1],
                                                                        rotations_splited_converted[2]，
                                                                        positions_splited_converted[0],
                                                                        positions_splited_converted[1],
                                                                        positions_splited_converted[2]
                                                                      );

You reversed the position and rotation，You should change it to：
         ignition::math::Pose3d jointOrigin = ignition::math::Pose3d(
                                                                        positions_splited_converted[0],
                                                                        positions_splited_converted[1],
                                                                        positions_splited_converted[2],
                                                                        rotations_splited_converted[0],
                                                                        rotations_splited_converted[1],
                                                                        rotations_splited_converted[2]
                                                                      );

**Comment by Ugo31** [comment]
Well seen thanks

**Comment by Ugo31** [comment]
Well seen, I had reversed it into my testing too so I couldn't see the problem... sorry 
It was probably the reason of @Timple and @mnbf9rca problems.

I link the corrected version to this post so they can check if it solve it !

[pegasus_gazebo_plugins_Gv9_v2.zip](https://github.com/wojiaojiao/pegasus_gazebo_plugins/files/3778918/pegasus_gazebo_plugins_Gv9_v2.zip)

**Comment by MCFurry** [comment]
Thx @Ugo31, except for the "had " on the first line of `closed_loop_plugin.cpp`, the rest works fine now!
Is this fix compatible with older Gazebo versions? I.e. would it be possible to merge these changes to this repo @Ugo31 @wojiaojiao ??

**Comment by Timple** [comment]
> I have the same issue as @mnbf9rca The links are seperated (by approx 1m in my simulation). They do seem to be connected.

The last commit in PR #4 actually solved the issue of the links being seperated.
I guess this PR is ready to be merged now :)

**Comment by nabihandres** [comment]
I try to install in melodic but i dont work when i put catkin_make it make a lot of errors

**Comment by RicardoDanielCPinto** [comment]
> I try to install in melodic but it don't work when i put catkin_make it make a lot of errors

Can't compile it either

Ubuntu 18 LTS, ROS melodic and Gazebo 9.0.0
It gave to much errors xD, but the first 2 were these:

```
/home/daniel/ws_moveit/src/pegasus_gazebo_plugins/src/closed_loop_plugin.cpp:1:5: error: stray ‘#’ in program
 had #include <pegasus_gazebo_plugins/closed_loop_plugin.h>
     ^
/home/daniel/ws_moveit/src/pegasus_gazebo_plugins/src/closed_loop_plugin.cpp:1:1: error: ‘had’ does not name a type; did you mean ‘char’?
 had #include <pegasus_gazebo_plugins/closed_loop_plugin.h>
 ^~~
 char
```

I just removed the "had", that is obviously wrong, and it compiled.
Thank you for your plugin.

**Comment by ycliu96** [comment]
How can I specify the joint type for the added joint？ @wojiaojiao
